stl/inc/memory

```diff
@@ -335,6 +440,36 @@ _NoThrowFwdIt destroy_n(_NoThrowFwdIt _First, const _Diff _Count_raw) {
     return _First;
 }
 
+#ifdef __cpp_lib_concepts
+namespace ranges {
+    // VARIABLE ranges::destroy_n
+    class _Destroy_n_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        // clang-format off
+        template <_No_throw_input_iterator _It>
+            requires destructible<iter_value_t<_It>>
+        /* _CONSTEXPR20_DYNALLOC */ _It operator()(_It _First, const iter_difference_t<_It> _Count) const noexcept {
+            // clang-format on
+            if (_Count <= 0) {
+                return _First;
+            }
+
+            auto _UFirst = _Get_unwrapped_n(_STD move(_First));
+            for (; _Count-- > 0; ++_UFirst) {

```

Comment:
Could we separate the decrement as we do in the other meow_n functions

I am totally fine with changing to this style but consistency

Reply:
I should probably `do`...`while` this like `std::destroy_n` since we know `_Count` is greater than `0` here. Actually I should steal the `is_trivially_destructible` optimization while I'm at it.

Reply:
I love me suggesting something tiny and inconsequential and you making things 10 times better in the process